src/queries/catalog/id-map-query/id-map-query-executor.ts

```diff
@@ -0,0 +1,21 @@
+import type { BasicQueryData } from '../../query';
+import { log } from '../../../util/log';
+import type { IdMapQuery, IdMapQueryResult } from './id-map-query-format';
+import { guard } from '../../../util/assert';
+
+
+export function executeIdMapQuery({ graph }: BasicQueryData, queries: readonly IdMapQuery[]): IdMapQueryResult {
+	if(queries.length !== 1) {
+		log.warn('Id-Map query expects only up to one query, but got', queries.length);
+	}
+
+	guard(graph.idMap !== undefined, 'The idMap is missing from the dataflow graph');
```

Comment:
if this is the case, please fallback to that in the normlaized AST, part of the query data,in general, just usee this directyl :)